aiida_common_workflows/workflows/relax/abinit/generator.py

```diff
@@ -120,12 +134,30 @@ def get_builder(
         inputs = generate_inputs(self.process_class._process_class, protocol, code, structure, override)  # pylint: disable=protected-access
         builder._update(inputs)  # pylint: disable=protected-access
 
-        if relaxation_type == RelaxType.ATOMS:
+        if relaxation_type == RelaxType.NONE:
+            optcell = 0
+            ionmov = 0
+        elif relaxation_type == RelaxType.ATOMS:
             optcell = 0
             ionmov = 22
+        elif relaxation_type == RelaxType.VOLUME:
+            optcell = 1
+            ionmov = 0
+        elif relaxation_type == RelaxType.SHAPE:
+            optcell = 3
+            ionmov = 0
+        elif relaxation_type == RelaxType.CELL:
+            optcell = 2
+            ionmov = 0
         elif relaxation_type == RelaxType.ATOMS_CELL:
             optcell = 2
             ionmov = 22
+        elif relaxation_type == RelaxType.ATOMS_VOLUME:
+            optcell = 1
+            ionmov = 22
+        elif relaxation_type == RelaxType.ATOMS_SHAPE:
+            optcell = 3
+            ionmov = 22
```

Comment:
Seems good. Have you tried (on Si) that all the combination worked ?

Reply:
I haven't tested all of the options, I'll write a quick script to make sure.

Reply:
Remove `VOLUME`, `CELL`, and `SHAPE`, which Abinit does not (easily) support.